fastn-p2p/src/cli/daemon/control.rs

```diff
@@ -0,0 +1,119 @@
+//! Control socket server for handling client requests
+//!
+//! This module handles the Unix domain socket that clients connect to.
+//! It parses JSON requests and coordinates with the P2P layer.
+
+use std::path::PathBuf;
+use tokio::sync::broadcast;
+use tokio::net::UnixListener;
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use serde::{Deserialize, Serialize};
+
+use super::{DaemonCommand, DaemonResponse};
+
+/// Client request types - precise typing for each operation
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type")]
+pub enum ClientRequest {
+    #[serde(rename = "call")]
+    Call {
+        from_identity: String,
```

Comment:
Let's make from_identity optional and let control server/daemon figure this out. We can always return error from control server, and this was cli does not need to see the internals of FASTN_HOME, only the path to control server unix socket.